File: src/main/java/com/agendarpgadmin/api/services/admin/LoginService.java
package com.agendarpgadmin.api.services.admin;
import java.util.UUID;

import com.agendarpgadmin.api.entities.UserEntity;
import com.agendarpgadmin.api.repositories.UserRepository;
import com.agendarpgadmin.api.services.utils.PasswordHashingService;
import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Date;

@Service
public class LoginService {

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private PasswordHashingService passwordHashingService;

    private final String SECRET_KEY = "secretaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
    private final long validityInMilliseconds = 86400000; // 1 dia

    public String authenticateUser(String email, String password) {
        UserEntity user = userRepository.findByEmail(email).orElse(null);
        if (user != null && passwordHashingService.verifyPassword(password, user.getPassword())) {
            // Verificar se o email foi verificado
            if (!Boolean.TRUE.equals(user.getEmailVerified())) {
                throw new IllegalStateException("Email não verificado");
            }
            return generateToken(user);
        }
        return null;
    }

    private String generateToken(UserEntity user) {
        Date now = new Date();
        Date validity = new Date(now.getTime() + validityInMilliseconds);
        return Jwts.builder()
                .setSubject(user.getEmail())
                .claim("nomeCompleto", user.getNomeCompleto())
                .claim("tipo", user.getTipo())
                .claim("id", user.getId())
                .setIssuedAt(now)
                .setExpiration(validity)
                .signWith(SignatureAlgorithm.HS256, SECRET_KEY)
                .compact();
    }
}

File: src/main/kotlin/com/agendarpgadmin/api/services/admin/LoginService.kt
package com.agendarpgadmin.api.services.admin

import com.agendarpgadmin.api.dtos.LoginResponseDTO
import com.agendarpgadmin.api.repositories.UserRepository
import com.agendarpgadmin.api.services.utils.JwtService
import com.agendarpgadmin.api.services.utils.PasswordHashingService
import org.springframework.stereotype.Service

/**
 * Serviço de Autenticação: O Guardião dos Portões da Agenda.
 * Gerencia o fluxo de login e emissão de tokens.
 */
@Service
class LoginService(
    private val userRepository: UserRepository,
    private val passwordHashingService: PasswordHashingService,
    private val jwtService: JwtService
) {

    /**
     * Autentica um usuário e gera o par de tokens (Access & Refresh).
     * 
     * @param email Email do aventureiro
     * @param password Senha em texto plano
     * @return DTO com tokens ou null se as credenciais forem heresia
     * @throws IllegalStateException Se o email ainda não foi purificado (verificado)
     */
    fun authenticateUser(email: String, password: String): LoginResponseDTO? {
        val user = userRepository.findByEmail(email).orElse(null) ?: return null

        if (passwordHashingService.verifyPassword(password, user.getPassword())) {
            // Um cavaleiro sem honra não entra na taverna
            if (user.getEmailVerified() != true) {
                throw IllegalStateException("Email não verificado")
            }

            val accessToken = jwtService.generateAccessToken(user)
            val refreshToken = jwtService.generateRefreshToken(user)

            return LoginResponseDTO(accessToken, refreshToken)
        }
        
        return null
    }
}
